app/controllers/comments_controller.rb

```diff
@@ -33,13 +31,21 @@ def update
   end
 
   def destroy
-    authorize :comment
-    current_user.comments.find(params[:id]).destroy
+    authorize comment
+    comment.destroy
     head :no_content
   end
 
   private
 
+  def comment
+    @comment ||= Comment.find(params[:id])
```

Comment:
You can query comments like this or you can query them through the book, like this book.comments.find(params[:id])